cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -0,0 +1,378 @@
+package appdetect
+
+import (
+	"fmt"
+	"log"
+	"maps"
+	"slices"
+	"strings"
+)
+
+type SpringBootProject struct {
+	springBootVersion     string
+	applicationProperties map[string]string
+	parentProject         *mavenProject
+	mavenProject          *mavenProject
+}
+
+type DatabaseDependencyRule struct {
+	databaseDep       DatabaseDep
+	mavenDependencies []MavenDependency
+}
+
+type MavenDependency struct {
+	groupId    string
+	artifactId string
+}
+
+var databaseDependencyRules = []DatabaseDependencyRule{
+	{
+		databaseDep: DbPostgres,
+		mavenDependencies: []MavenDependency{
+			{
+				groupId:    "org.postgresql",
+				artifactId: "postgresql",
+			},
+		},
+	},
+	{
+		databaseDep: DbMySql,
+		mavenDependencies: []MavenDependency{
+			{
+				groupId:    "com.mysql",
+				artifactId: "mysql-connector-j",
+			},
+		},
+	},
+	{
+		databaseDep: DbRedis,
+		mavenDependencies: []MavenDependency{
+			{
+				groupId:    "org.springframework.boot",
+				artifactId: "spring-boot-starter-data-redis",
+			},
+			{
+				groupId:    "org.springframework.boot",
+				artifactId: "spring-boot-starter-data-redis-reactive",
+			},
+		},
+	},
+	{
+		databaseDep: DbMongo,
+		mavenDependencies: []MavenDependency{
+			{
+				groupId:    "org.springframework.boot",
+				artifactId: "spring-boot-starter-data-mongodb",
+			},
+			{
+				groupId:    "org.springframework.boot",
+				artifactId: "spring-boot-starter-data-mongodb-reactive",
+			},
+		},
+	},
+	{
+		databaseDep: DbCosmos,
+		mavenDependencies: []MavenDependency{
+			{
+				groupId:    "com.azure.spring",
+				artifactId: "spring-cloud-azure-starter-data-cosmos",
+			},
+		},
+	},
+}
+
+func updateProjectByAnalyzingSpringBootProject(
+	parentProject *mavenProject, mavenProject *mavenProject, azdProject *Project) {
+	if !isSpringBootApplication(mavenProject) {
+		log.Printf("Skip analyzing spring boot project. path = %s.", mavenProject.path)
+		return
+	}
+	var springBootProject = SpringBootProject{
+		springBootVersion:     detectSpringBootVersion(parentProject, mavenProject),
+		applicationProperties: readProperties(azdProject.Path),
+		parentProject:         parentProject,
+		mavenProject:          mavenProject,
+	}
+	addDatabases(azdProject, &springBootProject)
+	addServiceBus(azdProject, &springBootProject)
+	addEventHubs(azdProject, &springBootProject)
+	addStorageAccount(azdProject, &springBootProject)
+	addSpringCloudAzure(azdProject, &springBootProject)
```

Comment:
Use one function for one service, this will make it much easier to add new service.